code/game/objects/structures/watercloset.dm

```diff
@@ -223,15 +223,47 @@
 	var/on = 0
 	var/obj/effect/mist/mymist = null
 	var/ismist = 0				//needs a var so we can make it linger~
-	var/watertemp = "normal"	//freezing, normal, or boiling
 	var/is_washing = 0
+	var/watertemp = "normal"	//freezing, normal, or boiling
 	var/list/temperature_settings = list("normal" = 310, "boiling" = T0C+100, "freezing" = T0C)
 
+	var/sound_id
+	var/datum/sound_token/sound_token
+
+//add heat controls? when emagged, you can freeze to death in it?
+
 /obj/structure/hygiene/shower/Initialize()
 	. = ..()
 	create_reagents(50)
+	sound_id = "[type]_[sequential_id(type)]"
```

Comment:
As I keep repeating, sound ids should be re-used to the greatest possible degree because the limit is 1024 unique ids.
`sound_id = /obj/structure/hygiene/shower` should suffice.